apps/explorer/src/pages/transaction-result/programmable-transaction-view/Command.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    type MoveCallSuiCommand,
+    type SuiArgument,
+    type SuiMovePackage,
+} from '@mysten/sui.js';
+import { type ReactNode } from 'react';
+
+import {
+    flattenSuiArguments,
+    stringifyCommandData,
+} from '~/pages/transaction-result/programmable-transaction-view/utils';
+import { ModuleLink, ObjectLink } from '~/ui/InternalLink';
+
+export interface ICommandProps<T> {
+    type: string;
+    data: T;
+}
+
+export enum CommandType {
+    SplitCoin = 'SplitCoin',
+    MergeCoins = 'MergeCoins',
+    TransferObjects = 'TransferObjects',
+    MoveCall = 'MoveCall',
+    Publish = 'Publish',
+    MakeMoveVec = 'MakeMoveVec',
+}
+
+function CommandContent({
+    type,
+    children,
+}: {
+    type: string;
+    children?: ReactNode;
+}) {
+    return (
+        <>
+            <div
+                data-testid="programmable-transactions-command-label"
+                className="text-heading6 font-semibold text-steel-darker"
+            >
+                {type}
+            </div>
+            {children && (
+                <div
+                    data-testid="programmable-transactions-command-content"
+                    className="text-bodyMedium pt-2 text-steel"
+                >
+                    {children}
+                </div>
+            )}
+        </>
+    );
+}
+
+function SplitMergeTransferMakeMoveVec({
+    type,
+    data,
+}: ICommandProps<(SuiArgument | SuiArgument[])[]>) {
+    return (
+        <CommandContent type={type}>
+            ({flattenSuiArguments(data)})
+        </CommandContent>
+    );
+}
+
+function MoveCall({ type, data }: ICommandProps<MoveCallSuiCommand>) {
+    const {
+        module,
+        package: movePackage,
+        function: func,
+        arguments: args,
+    } = data;
+    return (
+        <CommandContent type={type}>
+            (package: <ObjectLink objectId={movePackage} />, module:&nbsp;
+            <ModuleLink
+                objectId={movePackage}
+                moduleId={module}
+                label={module}
+            />
+            , function:&nbsp;
+            <span className="text-hero-dark">&#39;{func}&#39;</span>, arguments:
```

Comment:
Should probably be mono typeface and shouldn't be quote wrapped (design might disagree cc @mystie711)